### PR TITLE
Fix sporadic error on match query test when a fuzziness of 0s is used…

### PR DESCRIPTION
**Pull request by jimczi**
… on a date field.

### CODE REVIEW

**Comment by jimferenczi** [comment]
@cbuescher can you take a look ?

**Comment by cbuescher** [comment]
Great, I was also looking at this and this CI failure (http://build-us-00.elastic.co/job/es_core_master_oracle_6/5245/) right now.

File: core/src/test/java/org/elasticsearch/index/query/MatchQueryBuilderTests.java [hunk 1]
```diff
@@ -196,7 +196,7 @@ protected void doAssertLuceneQuery(MatchQueryBuilder queryBuilder, Query query,
             assertTrue(numericRangeQuery.includesMax());
 
             double value;
-            double width = 0;
+            double width = -1;
             try {
```
Comment (cbuescher) [inline core/src/test/java/org/elasticsearch/index/query/MatchQueryBuilderTests.java]: I think we can also get rid of the try/catch here, from the test setup we know its not date field if `queryBuilder.fieldName().equals(DATE_FIELD_NAME) == false`.

**Comment by cbuescher** [comment]
LGTM, left one tiny comment, maybe you want to add that as well.

**Comment by jimferenczi** [comment]
Thanks @cbuescher. I've changed the fix to include your comment. I agree it's cleaner to check the type of the field instead of the try/catch thingy.

**Comment by cbuescher** [comment]
@jimferenczi thanks, looks great.

**Pull request merged**